src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -91,7 +91,7 @@ func (r *ClusterReconciler) Reconcile(
 	var observedPods corev1.PodList
 
 	err := r.List(ctx, &observedPods, &client.ListOptions{
-		LabelSelector:	labels.SelectorFromSet(redpandaCluster.Labels),
+		LabelSelector:	labels.SelectorFromSet(getLabels(sts.LastObservedState, redpandaCluster.Labels)),
```

Comment:
I have a PR I had to close around the time we were merging the branches that will be in conflict with this. maybe that one actually resolves this as well?